src/coreclr/src/vm/class.cpp

```diff
@@ -1265,6 +1281,11 @@ void ClassLoader::ValidateMethodsWithCovariantReturnTypes(MethodTable* pMT)
             if (pMD == pParentMD)
                 continue;
 
+            // If the bit is not set on this method, but we reach here because it's been set on the method at the same slot on
+            // the base type, set the bit for the current method to ensure any future overriding method down the chain gets checked.
+            if (!pMD->RequiresCovariantReturnTypeChecking())
+                pMD->SetRequiresCovariantReturnTypeChecking();
```

Comment:
This is setting RequiresCovariantReturnTypeChecking too often. In particular, we should only force set the bit here if the parenteMD RequiresCovariantReturnTypeChecking.